server/cluster/cluster.go

```diff
@@ -0,0 +1,378 @@
+	if !exists {
+		return nil, ErrSchemaNotFound.WithCausef("schemaName", schemaName)
+	}
+
+	// Save table in storage.
+	tablePb := &clusterpb.Table{Id: tableID, Name: tableName, SchemaId: schema.GetID(), ShardId: shardID}
+	err := c.storage.CreateTable(ctx, c.clusterID, schema.GetID(), tablePb)
+	if err != nil {
+		return nil, errors.Wrap(err, "clusters CreateTable")
+	}
+
+	// Update tableCache in memory.
+	table := c.updateTableCacheLocked(schema, tablePb)
+	return table, nil
+}
+
+func (c *Cluster) loadClusterTopologyLocked(ctx context.Context) error {
+	clusterTopology, err := c.storage.GetClusterTopology(ctx, c.clusterID)
+	if err != nil {
+		return errors.Wrap(err, "clusters loadClusterTopologyLocked")
+	}
+	c.metaData.clusterTopology = clusterTopology
+
+	if c.metaData.clusterTopology == nil {
+		return ErrClusterTopologyNotFound.WithCausef("clusters:%v", c)
+	}
+
+	shardMap := make(map[uint32][]*clusterpb.Shard)
+	for _, shard := range c.metaData.clusterTopology.ShardView {
+		shardMap[shard.Id] = append(shardMap[shard.Id], shard)
+	}
+	c.metaData.shards = shardMap
+
+	shardIDs := make([]uint32, len(shardMap))
+	for id := range shardMap {
+		shardIDs = append(shardIDs, id)
+	}
+
+	c.metaData.shardIDs = shardIDs
+	return nil
+}
+
+func (c *Cluster) loadShardTopologyLocked(ctx context.Context, shardIDs []uint32) error {
+	topologies, err := c.storage.ListShardTopologies(ctx, c.clusterID, shardIDs)
+	if err != nil {
+		return errors.Wrap(err, "clusters loadShardTopologyLocked")
+	}
+	shardTopologyMap := make(map[uint32]*clusterpb.ShardTopology, len(c.metaData.shardIDs))
+	for i, topology := range topologies {
+		shardTopologyMap[c.metaData.shardIDs[i]] = topology
+	}
+	c.metaData.shardTopologies = shardTopologyMap
+	return nil
+}
+
+func (c *Cluster) loadSchemaLocked(ctx context.Context) error {
+	schemas, err := c.storage.ListSchemas(ctx, c.clusterID)
+	if err != nil {
+		return errors.Wrap(err, "clusters loadSchemaLocked")
+	}
+	schemaMap := make(map[string]*clusterpb.Schema, len(schemas))
+	for _, schema := range schemas {
+		schemaMap[schema.Name] = schema
+	}
+	c.metaData.schemas = schemaMap
+	return nil
+}
+
+func (c *Cluster) loadTableLocked(ctx context.Context) error {
+	for _, schema := range c.metaData.schemas {
+		tables, err := c.storage.ListTables(ctx, c.clusterID, schema.Id)
+		if err != nil {
+			return errors.Wrap(err, "clusters loadTableLocked")
+		}
+		for _, table := range tables {
+			if t, ok := c.metaData.tables[schema.GetName()]; ok {
+				t[table.GetId()] = table
+			} else {
+				c.metaData.tables[schema.GetName()] = map[uint64]*clusterpb.Table{table.GetId(): table}
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) GetSchema(schemaName string) (*Schema, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	schema, ok := c.schemasCache[schemaName]
+	return schema, ok
+}
+
+func (c *Cluster) GetTable(ctx context.Context, schemaName, tableName string) (*Table, bool, error) {
```

Comment:
`bool` is necessary?

Reply:
It is relatively clear to indicate whether the table exists. If use table!=nil, it is a bit strange.